src/TypeResolver.php

```diff
@@ -71,6 +71,7 @@ final class TypeResolver
     private $keywords = [
         'string' => Types\String_::class,
         'class-string' => Types\ClassString::class,
+        'trait-string' => Types\TraitString::class,
```

Comment:
What does `trait-string` provide that `class-string` does not? `class-string` is meant to be any class-like fqn.

Reply:
The goal is not to developp a new feature for PhpDocumentor as much as supporting a notation used by projects out there. If a user has trait-string in his codebase, it's not very useful that we already support class-string

Reply:
That doesn't answer the question. What is the difference? Why isn't there an interface-trait? Is class-string going to change?

Reply:
The author said in #111 the goal was to support Psalm's pseudo-types. Here's the list:
https://psalm.dev/docs/annotating_code/type_syntax/scalar_types/

Psalm doesn't support interface-string, so it's not in the list. No change is expected with class-string, trait-string is just a more narrow type that have been created for specific purposes in Psalm

Reply:
I think we are missing an interface in this lib. So there might be a small change to class-string. But it should not cause any bc breaks since we will just add the interface.

As said before there was a request to support more of psalms pseudo types. Which makes sense since this lib is reflecting the types in docblocks that people write out there. This library is not providing any functionality regarding the types we support.

Most of the supported pseudo types are just meta information to strings. Which do not even exist in php as a language.

Reply:
`class-string` resolves the fqsen, so it seems like `trait-string` should as well.

Reply:
No, by definition of https://psalm.dev/docs/annotating_code/type_syntax/scalar_types/ `trait-string` should not be parameterized like `class-string`. I opened a issue https://github.com/vimeo/psalm/issues/3865 to clarify how `trait-string` should be used and @orklah has already provided an example https://psalm.dev/r/3bff912a5b

Reply:
Not everything is about implementing psalm types. Since `class-string` is essentially the same as `trait-string`, it makes sense that they would behave similar in TypeResolver. That's what I'm saying.

Reply:
@jaapio How I should name the interface? `StringType` or `StringableType`?

Reply:
Just dumping some thoughts here:
- `StaticString` because they are representing a value of `String_` those types are not really a Type.  It is a `String_` but with a certain value type. 
- `TypedString`
- `StringRepresentation`
 
- `StringType` should be avoided I think, because it implicates that it is a type.. 

Maybe they should even extend `String_` But I was trying to avoid that, because it would open doors for a large inheritance tree. Which I don't like.